examples/nodejs/token-vending-machine/infrastructure/lib/token-vending-machine-stack.ts

```diff
@@ -0,0 +1,51 @@
+import * as path from 'path';
+import * as cdk from 'aws-cdk-lib';
+import {Construct} from 'constructs';
+import * as lambda from 'aws-cdk-lib/aws-lambda';
+import * as lambdaNodejs from 'aws-cdk-lib/aws-lambda-nodejs';
+import * as apig from 'aws-cdk-lib/aws-apigateway';
+import * as secrets from 'aws-cdk-lib/aws-secretsmanager';
+
+export class TokenVendingMachineStack extends cdk.Stack {
+  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
+    super(scope, id, props);
+
+    const momentoAuthTokenParam = new cdk.CfnParameter(this, 'MomentoAuthTokenParam', {
+      type: 'String',
+      description:
+        'The Momento Auth Token that will be used to vend browser tokens. Will be stored in Secrets Manager.',
+      noEcho: true,
+    });
+
+    const authTokenSecret = new secrets.Secret(this, 'MomentoTokenVendingMachineAuthToken', {
+      secretName: 'MomentoTokenVendingMachineAuthToken',
+      secretStringValue: new cdk.SecretValue(momentoAuthTokenParam.valueAsString),
+    });
+
+    const tvmLambda = new lambdaNodejs.NodejsFunction(this, 'MomentoTokenVendingMachine', {
+      functionName: 'MomentoTokenVendingMachine',
+      runtime: lambda.Runtime.NODEJS_16_X,
+      entry: path.join(__dirname, '../../lambda/handler.ts'),
+      projectRoot: path.join(__dirname, '../../lambda'),
+      depsLockFilePath: path.join(__dirname, '../../lambda/package-lock.json'),
+      handler: 'handler',
+      timeout: cdk.Duration.seconds(30),
+      memorySize: 128,
```

Comment:
This is async job not blocking user experience at all right?

Reply:
no, it's not async. the browser makes a request to the APIG to get a temporary token before it can start making requests for topics.

Reply:
I think we should maybe bump this up if hit directly by user

Reply:
okay. for a nodejs lambda that is doing such a trivial amount of work i'm not sure it will make any difference but i'm happy to bump it up. what number are you thinking

Reply:
I would just run quick comparison go up to 1GB or 2GB quick see if makes big difference. If its trivial then fine keeping at 128.

Reply:
@poppoerika I may need to leave this part as an exercise for you. we can discuss on our call.

Reply:
totally fine for follow up pr IMO